flocker/testtools/__init__.py

```diff
@@ -631,7 +633,7 @@ def remove_image():
                         client.remove_container(container[u"Names"][0])
                 client.remove_image(tag, force=True)
             self.test.addCleanup(remove_image)
-        return tag
+        return d.addCallback(lambda ignored: tag)
```

Comment:
Change `build` docstring to indicate it now returns a Deferred containing the bytes with the tag name